### DISCUSSION TITLE
Web based color Generator / Palette

### DISCUSSION THREAD

**Discussion by kamihicouki** [tailwindlabs/tailwindcss#2405] [Ideas]
# Situation
* I'm a front engineer and a web designer works somewhere far away.
* I like color codes and someone far away likes visual colors.

So I thought.
"It would be nice if Tailwind's site had color palette creation and color code generation capabilities like Ionic."

 - [Customizing Colors -Tailwind CSS] (https://tailwindcss.com/docs/customizing-colors) 
 - [Color Generator -Ionic Documentation] (https://ionicframework.com/docs/theming/color-generator/)

**Comment by csenio** [comment]
The math for this is hell :D I think lyft comes closest to sth like that with https://www.colorbox.io/

**Comment by ThePlayerSD** [comment]
I personally use https://javisperez.github.io/tailwindcolorshades/#/?aqua=207376&tv=1 for the Colours

Leave it blank and it will generate an interesting name for the colour for you!

Hope this helps!

**Comment by samzlab** [comment]
Just released a [color-generator lib](https://www.npmjs.com/package/tailwind-hsluv) for Tailwind CSS. It's using the HSLuv color space for better lightness distribution.

I would like to make a plugin from this lib, but didn't figured out yet how its possible without directly touch the config object. As far as i know there is no function for extend/override the colors from a plugin.

**Reply by adamwathan** [reply]
You can actually! The plugin function takes a second argument which is your own config object you'd like merged in:

https://tailwindcss.com/docs/plugins#providing-default-options

**Reply by samzlab** [reply]
So far I just thought of it as it is for defaults only - until now :) I will play with it tomorrow :) Thank you Adam!

**Comment by kamihicouki** [comment]
Hello and good evening.
Thank you for all the replies and discussions.

I have just discovered a very amazing project.
I still don't know what the project is about, but it seems to be very close to the tool I was looking for.

I'm happy to share it with you.

https://github.com/dmarman/dmarman.github.io

**Comment by tombohub** [comment]
@kamihicouki  if it helps, Ive made color palette generator for tailwind. https://palettolithic.com
Let me know if you have ideas to improve
